docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+
+#### 4.3 Function Calls
+
+Functions take in arguments by value except in the case of other functions which are passed by reference. Functions need to be assigned before being called but can be declared anonymously. An anonymous function's scope is within the function that it is declared.
+
+#### 4.4 Logical Negation
+
+TODO We're considering adding booleans, its easier to parse.
+
+Although FIRE does not directly support a boolean type, it equates `true` expressions to `1` and false expressions to `0`. As such, the logical negation operator `!` will convert the result of the logical expression on which it is applied to the inverse of what would normally be expected.
+
+All non-zero integers are evaluated as `true` where as `0` is false.
+
+#### 4.5 Logical AND Operator
+
+The logical AND `&&` is a short circuit operator, and returns 1 if and only if the expressions on its left and right both evaluate to 1.
+
+#### 4.6 Logical OR Operator
+
+The logical OR `||` operator is a short circuit operator and returns 1 if either of the expressions on its left or right return 1.
+
+#### 4.7 Relational Operators
+
+The relational operators `<, >, <=, <=, ==` return `1` if the expression on the left side of the operator has the expected relation to the operator on the right-hand side.
+
+These relationships amongst ints are determined by natural ordering. Strings can only be evaluated using the `==` operator.
+
+#### 4.8 Pattern Match Operator
+
+The pattern match operator `===` returns `1` if the regular expression or string on its right side conforms to the rules laid out by the regular expression on its left side and returns `0` otherwise.
+
+Pattern matching operator l-values must be type `regx` and r-values must be type `string`.
+
+Example: `r'[a-z]+' === word`
+
+#### 4.9 String Concatination Operator
+
+The string concatenation `^` operator returns a new string that is the concatenation of the string on its left side and the string on its right side.
+
+#### 4.10 Bracket Operator
+
+The bracket operator `[]` can be used with either`string` or `array`.
+
+When used on`array` it is supplied a key and returns the corresponding element, or `-1` if it does not exist.
+
+When used on `string` it functions in a similar manner to character arrays in C. The r-value in brackets is an integer and returns the letter of that index, however unlike C, FIRE does not support chars so it returns a string of length 1.
+
+```
+str hello = "hello";
+str o = hello[4]; 
+```
+#### 4.11 Slice operator 
+
+The slice operator `[x:y]` is used on a string and returns a substring.
+
+#### 4.12 Map 
+
+The map keyword allows a programmer to apply a function to every element of an array. 
+
+Example: `map(f,arr1);` 
+
+The map keyword does not mutate the values in the provided array; it instead returns a new array with results of every element of arr1 after they are passed to func f. 
+
```

Comment:
what are the typing constraints?

Reply:
and what if `f` returns `void`? is that allowed? if so, what's the type of the array?

Reply:
Expanded